### PR TITLE
Enhance dml channel operations

### PR DESCRIPTION
**Pull request by cydrain**
Signed-off-by: yudong.cai <yudong.cai@zilliz.com>

### CODE REVIEW

File: internal/rootcoord/dml_channels.go [hunk 1]
```diff
@@ -82,13 +82,12 @@ func (d *dmlChannels) GetNumChannels() int {
...
-				return err
+		if _, ok1 := d.refcnt.Load(chanName); ok1 {
+			if v, ok2 := d.pool.Load(chanName); ok2 {
+				if err := (*(v.(*msgstream.MsgStream))).Broadcast(pack); err != nil {
+					return err
+				}
 			}
-		} else {
-			return fmt.Errorf("channel %s not exist", chanName)
 		}
```
Comment (congqixia) [inline internal/rootcoord/dml_channels.go:91]: If the channel name is not valid, an error should be returned. I suggest that an error returns if either map fails to load with provided channel name.

File: internal/rootcoord/dml_channels.go [hunk 2]
```diff
@@ -36,40 +42,43 @@ func newDmlChannels(c *Core, chanNamePrefix string, chanNum int64) *dmlChannels
...
-	cnt := d.idx.Load()
-	name := fmt.Sprintf("%s_%d", d.namePrefix, cnt)
-	d.idx.Store((cnt + 1) % d.capacity)
-	return name
+	cnt := d.idx.Inc()
+	return getDmlChannelName(d.namePrefix, (cnt-1)%d.capacity)
 }
 
 // ListChannels lists all dml channel names
 func (d *dmlChannels) ListChannels() []string {
```
Comment (xiaofan-luan) [inline internal/rootcoord/dml_channels.go:73]: We should name it to ListPysical channels?

Comment (cydrain) [reply internal/rootcoord/dml_channels.go]: done

**Pull request merged**